src/modules/AutoinstPartPlan.ycp

```diff
@@ -349,6 +354,59 @@
   }
 
 
+
+  /**
+   * Software raids on multipath.
+   * The device names in the device order are mapped to e.g.
+   * /dev/dm-4. This is wrong for multipath devices which still need
+   * mapped devices like /dev/mapper/<number>_part2.
+   * So, we are switching back in order to write the correct
+   * AutoYaST settings. (bnc#996839)
+   */
+
+  define list map_device_order(list<string> devices)
+  {
+    list<string> mapped_devices = [];
+    foreach( string device, devices,
+	     ``{
+	       if( search( device, "/dev/dm-" ) == 0 )
+	       {
+		 if( size(DiskRaids) == 0 )
```

Comment:
Could you get this condition out of the loop? I guess it's something you only need to do one time, right? (maybe I'm missing something in this YCP code) :)